wp-content/plugins/core/src/Blocks/Block_Base.php

```diff
@@ -92,4 +93,30 @@ public function enqueue_block_style(): void {
 		] );
 	}
 
+	public function register_admin_scripts(): void {
+		$block    = $this->get_block_handle();
+		$path     = $this->get_block_path();
+		$args     = $this->get_asset_file_args( get_theme_file_path( "dist/blocks/$path/admin.asset.php" ) );
+		$src_path = get_theme_file_path( "dist/blocks/$path/admin.js" );
+		$src      = get_theme_file_uri( "dist/blocks/$path/admin.js" );
+
+		if ( ! file_exists( $src_path ) ) {
+			return;
+		}
+
+		wp_register_script(
+			"admin-$block-scripts",
+			$src,
+			$args['dependencies'] ?? [],
+			$args['version'] ?? false,
+			true
+		);
+	}
+
+	public function enqueue_admin_scripts(): void {
+		$block = $this->get_block_handle();
+
+		wp_enqueue_script( "admin-$block-scripts" );
+	}
+
```

Comment:
These functions enqueue the new `admin.js` files from `dist` with a new `file_exists` check instead of using a remote get function.